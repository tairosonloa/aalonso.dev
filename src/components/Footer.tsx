import React, { FC } from 'react'
import ScrollToTop from '../containers/scrollToTop'
import SocialMediaLinks from './Icons/Icons'

const Footer: FC = () => {
  return (
    <div className="footer w-full mt-8 bg-gradient-to-b from-transparent to-gray-800">
      <ScrollToTop />
      <footer className="flex flex-col sm:flex-row sm:w-11/12 md:w-3/4 lg:w-2/4 m-auto sm:justify-between items-center w-full">
        <SocialMediaLinks />
        <p className="text-sm mb-6 sm:mb-4 text-gray-400 sm:self-center">
          &copy; 2020-{new Date().getFullYear()} Aitor Alonso
        </p>
      </footer>
    </div>
  )
}

export default Footer
