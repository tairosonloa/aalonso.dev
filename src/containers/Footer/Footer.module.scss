@import 'src/styles/utils.scss';
@import 'src/styles/palette.scss';

.container {
  display: grid;
  position: relative;
  grid-gap: grid_units(4);
  grid-template-columns: repeat(auto-fit, minmax(grid_units(45), 1fr));
  padding: grid_units(4);
  background-color: $rich-black;

  & > * {
    display: flex;
    align-items: center;
    justify-content: center;
    color: $light-gray;
  }

  @include screenIn(smallDesktop) {
    & > :nth-child(2) {
      justify-content: flex-start;
    }

    & > :last-child {
      justify-content: flex-end;
    }
  }
}

.copyright {
  margin: 0;
}

.socialMediaLinks i,
.link {
  color: $light-gray;
}
