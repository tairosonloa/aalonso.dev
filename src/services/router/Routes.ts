export class Route {
  private readonly path: string

  static of(path: string): Route {
    return new Route(path)
  }

  private constructor(path: string) {
    this.path = path
  }

  getPath(): string {
    return this.path
  }
}

export const Routes = {
  privacyPolicy: (): Route => Route.of('/privacy-policy'),
}
