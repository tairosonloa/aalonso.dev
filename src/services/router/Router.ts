import { Route } from './Routes'

export interface Router {
  replaceWith(route: Route): Promise<void>
  pushTo(route: Route): Promise<void>
}
