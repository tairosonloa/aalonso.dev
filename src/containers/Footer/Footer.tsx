import React, { FC } from 'react'
import { Copyright } from '../../components/Copyright/Copyright'
import { ScrollUpButton } from '../../components/ScrollUpButton/ScrollUpButton'
import { SocialMediaLinks } from '../../components/SocialMediaLinks/SocialMediaLinks'
import { Router } from '../../services/router/Router'
import { Routes } from '../../services/router/Routes'
import styles from './Footer.module.scss'

type FooterProps = {
  router: Router
}

export const Footer: FC<FooterProps> = ({ router }) => {
  return (
    <footer className={styles.container}>
      <ScrollUpButton />
      <Copyright className={styles.copyright} />
      <SocialMediaLinks className={styles.socialMediaLinks} />
      <div>
        <a
          className={styles.link}
          href={Routes.privacyPolicy().getPath()}
          onClick={(e) => {
            e.preventDefault()
            router.pushTo(Routes.privacyPolicy())
          }}
        >
          Privacy & Cookies
        </a>
      </div>
    </footer>
  )
}
