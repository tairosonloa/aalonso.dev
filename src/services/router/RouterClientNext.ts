import { NextRouter } from 'next/router'
import { Router } from './Router'
import { Route } from './Routes'

export class RouterClientNext implements Router {
  private router: NextRouter

  constructor(router: NextRouter) {
    this.router = router
  }

  async pushTo(route: Route): Promise<void> {
    await this.router.push(route.getPath())
  }

  async replaceWith(route: Route): Promise<void> {
    await this.router.replace(route.getPath())
  }
}
