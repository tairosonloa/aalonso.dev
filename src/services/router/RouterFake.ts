import noop from 'lodash/noop'
import { Router } from './Router'
import { Route } from './Routes'

type Options = {
  /**
   * Push spy
   */
  push: (path: string) => void
  /**
   * Replace spy
   */
  replace: (path: string) => void
}

export class RouterFake implements Router {
  private readonly push: (path: string) => void

  private readonly replace: (path: string) => void

  constructor({ push = noop, replace = noop }: Partial<Options> = {}) {
    this.push = push
    this.replace = replace
  }

  async pushTo(route: Route): Promise<void> {
    this.push(route.getPath())
  }

  async replaceWith(route: Route): Promise<void> {
    this.replace(route.getPath())
  }
}
