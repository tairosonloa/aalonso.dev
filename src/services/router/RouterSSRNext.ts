import { GetServerSidePropsContext } from 'next'
import { Router } from './Router'
import { Route } from './Routes'

export class RouterSSRNext implements Router {
  private ctx: GetServerSidePropsContext

  constructor(ctx: GetServerSidePropsContext) {
    this.ctx = ctx
  }

  getPropertyId(): string | undefined {
    return this.ctx?.params?.propertyId as string
  }

  async pushTo(route: Route): Promise<void> {
    this.ctx.res.writeHead(302, { Location: route.getPath() })
    this.ctx.res.end()
  }

  async replaceWith(route: Route): Promise<void> {
    await this.pushTo(route)
  }
}
