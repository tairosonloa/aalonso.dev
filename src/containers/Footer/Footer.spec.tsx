import { act, fireEvent, render } from '@testing-library/react'
import * as React from 'react'
import { RouterFake } from '../../services/router/RouterFake'
import { Footer } from './Footer'

const pushSpy = jest.fn()
const routerSpy = new RouterFake({ push: pushSpy })

describe('<Footer/>', () => {
  it('should display the default message', () => {
    const renderResult = render(<Footer router={routerSpy} />)

    expect(renderResult.getByText('Copyright', { exact: false })).toBeTruthy()
    expect(renderResult.getByText('Privacy & Cookies')).toBeTruthy()
  })

  describe('Footer links', () => {
    it('should navigate to privacy policy page', async () => {
      const renderResult = render(<Footer router={routerSpy} />)

      const privacyLink = renderResult.getByText('Privacy & Cookies')
      await act(async () => {
        fireEvent.click(privacyLink)
      })
      expect(pushSpy).toHaveBeenCalledWith('/privacy-policy')
    })
  })
})
