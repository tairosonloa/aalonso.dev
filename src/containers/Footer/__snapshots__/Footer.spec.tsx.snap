// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="container"
      >
        <button
          class="scrollTop"
          data-testid="scrollUpButton"
          type="button"
        >
          <i
            class="ri-arrow-up-line icon"
          />
        </button>
        <p
          class="common paragraph copyright"
        >
          Copyright © 
          2020
           
          Aitor Alonso
        </p>
        <ul
          class="container socialMediaLinks"
        >
          <li>
            <a
              data-testid="github-link"
              href="https://github.com/tairosonloa"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="ri-github-fill icon"
                data-testid="github"
              />
              <span
                class="hiddenTextForAccessibility"
              >
                github
              </span>
            </a>
          </li>
          <li>
            <a
              data-testid="linkedin-link"
              href="https://www.linkedin.com/in/aitoralonsonunez/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="ri-linkedin-box-fill icon"
                data-testid="linkedin"
              />
              <span
                class="hiddenTextForAccessibility"
              >
                linkedin
              </span>
            </a>
          </li>
          <li>
            <a
              data-testid="twitter-link"
              href="https://twitter.com/tairosonloa"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="ri-twitter-fill icon"
                data-testid="twitter"
              />
              <span
                class="hiddenTextForAccessibility"
              >
                twitter
              </span>
            </a>
          </li>
          <li>
            <a
              data-testid="email-link"
              href="mailto:aitor@aalonso.dev"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="ri-mail-fill icon"
                data-testid="email"
              />
              <span
                class="hiddenTextForAccessibility"
              >
                email
              </span>
            </a>
          </li>
        </ul>
        <div>
          <a
            class="link"
            href="/privacy-policy"
          >
            Privacy & Cookies
          </a>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="container"
    >
      <button
        class="scrollTop"
        data-testid="scrollUpButton"
        type="button"
      >
        <i
          class="ri-arrow-up-line icon"
        />
      </button>
      <p
        class="common paragraph copyright"
      >
        Copyright © 
        2020
         
        Aitor Alonso
      </p>
      <ul
        class="container socialMediaLinks"
      >
        <li>
          <a
            data-testid="github-link"
            href="https://github.com/tairosonloa"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="ri-github-fill icon"
              data-testid="github"
            />
            <span
              class="hiddenTextForAccessibility"
            >
              github
            </span>
          </a>
        </li>
        <li>
          <a
            data-testid="linkedin-link"
            href="https://www.linkedin.com/in/aitoralonsonunez/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="ri-linkedin-box-fill icon"
              data-testid="linkedin"
            />
            <span
              class="hiddenTextForAccessibility"
            >
              linkedin
            </span>
          </a>
        </li>
        <li>
          <a
            data-testid="twitter-link"
            href="https://twitter.com/tairosonloa"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="ri-twitter-fill icon"
              data-testid="twitter"
            />
            <span
              class="hiddenTextForAccessibility"
            >
              twitter
            </span>
          </a>
        </li>
        <li>
          <a
            data-testid="email-link"
            href="mailto:aitor@aalonso.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="ri-mail-fill icon"
              data-testid="email"
            />
            <span
              class="hiddenTextForAccessibility"
            >
              email
            </span>
          </a>
        </li>
      </ul>
      <div>
        <a
          class="link"
          href="/privacy-policy"
        >
          Privacy & Cookies
        </a>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
